File: chor-extraction/src/main/java/ast/sp/nodes/SelectionSP.java
package ast.sp.nodes;

import ast.sp.interfaces.Behaviour;
import ast.sp.interfaces.Interaction;

public class SelectionSP implements Interaction {
    private final Behaviour continuation;
    private final String process;
    private final String expression;

    public SelectionSP(Behaviour continuation, String process, String expression) {
        this.continuation = continuation;
        this.process = process;
        this.expression = expression;
    }

    public Behaviour getContinuation() {
        return continuation;
    }

    public String getProcess() {
        return process;
    }

    public String getExpression() {
        return expression;
    }

    public String toString() {
        return process + " + " + expression + "; " + continuation.toString();
    }

    @Override
    public boolean findRecProcCall(String procname) {
        return continuation.findRecProcCall(procname);
    }
}

File: chor-extraction/src/main/java/ast/sp/nodes/SelectionSP.kt
package ast.sp.nodes

import ast.sp.interfaces.Behaviour
import ast.sp.interfaces.Interaction

data class SelectionSP(val continuation: Behaviour, val process: String, val expression: String) : Interaction {

    override fun toString(): String {
        return process + " + " + expression + "; " + continuation.toString()
    }

    override fun findRecProcCall(procname: String): Boolean {
        return continuation.findRecProcCall(procname)
    }

    override fun copy(): Interaction {
        return SelectionSP(continuation.copy(), process, expression)
    }

    override fun equals(b: Behaviour): Boolean {
        return b is SelectionSP &&
                b.process == process &&
                b.expression == expression &&
                b.continuation.equals(continuation)
    }
}
